src/GraphErrorHandler.ts

```diff
@@ -57,25 +57,17 @@ export class GraphErrorHandler {
 	 *      }
 	 *  }
 	 */
-	private static constructErrorFromResponse(error: any, statusCode: number): GraphError {
-		if (!error.error) {
-			// example - Consider error specified is a string when using type AuthProviderCallback = (error: any, accessToken: string | null) => void;
-			const graphError = new GraphError(statusCode, "Microsoft Graph JavaScript Client SDK came across an error while processing the request");
-			graphError.body = error;
-			return graphError;
-		}
-		error = error.error;
+	private static constructErrorFromResponse(graphError: any, statusCode: number): GraphError {
```

Comment:
add a type to avoid any and get transpile time checks?

Reply:
Can you please elaborate on transpile time checks?

Reply:
right now, because graphError is of any type, none of the things referenced under the graphError parameter are being checked by the compiler/transpiler.
Effectively if somebody renames error to errorObject or errorPayload in the source object that's being passed here, this method will break.
Not only it will break, but it won't say anything at compile time and break at runtime.
Using any also doesn't help consumers as they don't know what this method is expecting.

Reply:
1. This is a private method which is specifically designed around the Graph API Error response. I agree that the code should be loosely coupled. I will create an issue after finding  more on this.
2. The error passed to the GraphErrorHandler could be a Graph API Error or could also be any custom error (say if  using a custom middleware).  GraphErrorHandler has been taking into consideration both a Graph API Error and any custom error.

Reply:
if we don't want to replace any by something else here for the reasons you have exposed, we should at least test the properties in use before using them to avoid errors at runtime.
Things like testing if graphError.error exists before assigning it (defensive programming) or using null propagation operators (graphError && graphError.error).
Another alternative would be to replace any by a prototype of what that method accesses i.e. something like that
`{error: {message: string, code: number}}...`